File: abc/abc179/abc179_b.cs
// https://atcoder.jp/contests/abc179/submissions/16890189

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var D = new int[N][];
        for (int i = 0; i < N; i++)
        {
            D[i] = new int[2];
            D[i] = Console.ReadLine().Split().Select(int.Parse).ToArray();
        }
        var count = 0;
        var result = "No";
        for (int i = 0; i < N; i++)
        {
            if (D[i][0] == D[i][1])
            {
                count++;
            }
            else
            {
                count = 0;
            }
            if (count == 3)
            {
                result = "Yes";
                break;
            }
        }
        Console.WriteLine(result);
    }
}

File: abc/abc179/abc179_b.fs
// https://atcoder.jp/contests/abc179/submissions/16890570

let N = stdin.ReadLine() |> int

let D =
    [ for i in [ 0..N - 1 ] do
        yield (stdin.ReadLine().Split() |> Array.map int) ]

let rec solver n i =
    if i = N then "No"
    elif D.[i].[0] = D.[i].[1] then
        if n + 1 = 3 then "Yes"
        else solver (n + 1) (i + 1)
    else solver 0 (i + 1)

solver 0 0 |> stdout.WriteLine
